### PR TITLE
Add code highlighting and "Copy to Clipboard" for config, along with inline comments in config

### PR DESCRIPTION
**Pull request by DrewML**
## This PR is a:

- [ ] New feature
- [X] Enhancement/Optimization
- [ ] Refactor
- [ ] Bugfix
- [ ] Test for existing code
- [ ] Documentation

### CODE REVIEW

File: src/JavaScriptHighlighter/JavaScriptHighlighter.css [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
```
Comment (jedmao) [inline src/JavaScriptHighlighter/JavaScriptHighlighter.css]: Can you use the [Webpack `BannerPlugin`](https://webpack.js.org/plugins/banner-plugin/) for this?

File: src/RequireJS/BundleGenerator/ConfigGen.tsx [hunk 2]
```diff
@@ -6,17 +6,74 @@
...
 };
 
-export default class ConfigGen extends React.Component<Props> {
-    render() {
+export default class ConfigGen extends React.PureComponent<Props> {
+    generateJS() {
         const { pageModules, requireConfig } = this.props;
         const config = generate(pageModules, requireConfig);
+        const code = JSON.stringify(config, null, 4);
+        return transforms.reduce((code: string, transform) => {
```
Comment (jedmao) [inline src/RequireJS/BundleGenerator/ConfigGen.tsx]: I don't know if you already know this or not, but if you're immediately returning something, you can just reduce this to:
```tsx
return transforms.reduce((_code, transform) => transform(_code), code);
```
Notice the following too:
1. If `transforms` is properly typed, you don't need to indicate the type of `code: string` in the callback.
2. `code: string` was shadowing the existing `const code` defined on the line just above. I fixed that with a leading underscore. I normally have lint rules against shadowing, FWIW.

File: src/RequireJS/BundleGenerator/ConfigGen.tsx [hunk 3]
```diff
@@ -6,17 +6,74 @@
...
+    const [whitespace] = byLine[sharedBundleIndex].match(/(\s+)/) || [''];
+    byLine.splice(
+        sharedBundleIndex,
+        0,
+        `${whitespace}// Modules used on > 1 page(s) of the store`,
+    );
+    return byLine.join('\n');
+}
+
+function uglifyComment(code: string) {
```
Comment (jedmao) [inline src/RequireJS/BundleGenerator/ConfigGen.tsx]: This function shares a lot of the same logic as `sharedBundleComment`. Recommend consolidation.

File: src/copyToClipboard.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+export default function copyToClipboard(text: string) {
+    const el = document.createElement('textarea');
+    el.value = text;
+    el.style.position = 'absolute';
+    el.style.left = '-9999px';
+
+    document.body.appendChild(el);
+    el.select();
+
+    let success;
```
Comment (jedmao) [inline src/copyToClipboard.ts]: You can avoid creating a `success` variable by returning the success value directly and removing the child in a `finally` block:
```ts
try {
    return document.execCommand('copy');
} catch (err) {
    console.error(err);
    return false;
} finally {
    document.body.removeChild(el);
}
```

File: src/index.d.tsx [hunk 5]
```diff
@@ -7,3 +7,8 @@ declare module '*.css' {
     const content: { [className: string]: string };
     export default content;
 }
+
+declare module 'highlight-javascript-syntax' {
```
Comment (jedmao) [inline src/index.d.tsx]: I was super surprised to find that `npm install @types/highlight-javascript-syntax` wasn't found. 🤷‍♂️

Comment (DrewML) [reply src/index.d.tsx]: Yeah, not exactly a popular one. Just super simple

File: src/copyToClipboard.ts (hunk 4 above)
Comment (DrewML) [reply src/copyToClipboard.ts]: Yeah this was like 99% copy pasta from a Sindre module that had extra cruft I didn't want for mobile devices 😝

File: src/RequireJS/BundleGenerator/ConfigGen.tsx (hunk 3 above)
Comment (DrewML) [reply src/RequireJS/BundleGenerator/ConfigGen.tsx]: Big big big proponent of the [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming))

File: src/RequireJS/BundleGenerator/ConfigGen.tsx (hunk 2 above)
Comment (DrewML) [reply src/RequireJS/BundleGenerator/ConfigGen.tsx]: > If transforms is properly typed, you don't need to indicate the type of code: string in the callback.

Whoops! Was dealing with some type errors and temporarily added that to debug.

> code: string was shadowing the existing const code defined on the line just above. I fixed that with a leading underscore. I normally have lint rules against shadowing, FWIW.

Yep, I knowingly did that. Was the most sensical name for the identifier I could come up with at the time, and didn't need to access `code` in the outer scope.

> I don't know if you already know this or not, but if you're immediately returning something, you can just reduce this to:

Yep! I use implicit return plenty, but in this case the implicit return version blew the line limit for prettier's formatting, and I thought it looked a bit more readable spread out.

File: src/index.d.tsx (hunk 5 above)
Comment (jedmao) [reply src/index.d.tsx]: I didn't know they had some popularity threshold on that. I thought Microsoft had some sort of bot that published `@types`.

File: src/JavaScriptHighlighter/JavaScriptHighlighter.css (hunk 1 above)
Comment (DrewML) [reply src/JavaScriptHighlighter/JavaScriptHighlighter.css]: That would be much nicer. I was asked to have the banner in every _source_ file, though.

File: src/RequireJS/BundleGenerator/ConfigGen.tsx (hunk 3 above)
Comment (jedmao) [reply src/RequireJS/BundleGenerator/ConfigGen.tsx]: ⚾️

File: src/index.d.tsx (hunk 5 above)
Comment (DrewML) [reply src/index.d.tsx]: > I didn't know they had some popularity threshold on that. I thought Microsoft had some sort of bot that published @types.

Whoa really - I thought the entirety of definitely-typed was manually created?

Comment (jedmao) [reply src/index.d.tsx]: DefinitelyTyped is manual, but I thought `@types` covered more than just what's available on DT.

**Pull request merged**